src/WebUI/Components/ListComponent.tsx

```diff
@@ -3,56 +3,47 @@
     Licensed under the MIT license.
 */
 
-import { IColumn } from "azure-devops-ui/Components/VssDetailsList/VssDetailsList.Props";
 import { TooltipHost, TooltipOverflowMode } from "azure-devops-ui/Tooltip";
-import { VssDetailsList } from "azure-devops-ui/VssDetailsList";
-import { ConstrainMode, SelectionMode, IDetailsHeaderProps, IDetailsRowProps, DetailsRow } from "office-ui-fabric-react/lib/DetailsList";
 import { BaseComponent, css, IRenderFunction } from "office-ui-fabric-react/lib/Utilities";
 import * as React from "react";
 import { IVssComponentProperties } from "../Types";
+import { Table, ITableColumn, TableRow, ITableRowProps, SimpleTableCell } from "azure-devops-ui/Table";
+import { ITableRow, ITableRowDetails } from "azure-devops-ui/Components/Table/Table.Props";
+import { ArrayItemProvider } from "azure-devops-ui/Utilities/Provider";
 import "./ListComponent.scss";
 
 export interface IListComponentProperties<T> extends IVssComponentProperties {
     headingText?: string;
     headingContent?: JSX.Element;
     items: T[];
-    columns: IColumn[];
-    onRenderItemColumn: (item?: T, index?: number, column?: IColumn) => React.ReactNode;
-    onItemInvoked?: (item?: any, index?: number, ev?: Event) => void;
+    columns: ITableColumn<T>[];
+    onItemActivated?: (event: React.SyntheticEvent<HTMLElement>, tableRow: ITableRow<any>, selectedItem: any) => void;
 }
 
 export class ListComponent<T> extends BaseComponent<IListComponentProperties<T>> {
     public render(): React.ReactNode {
         return (
             <div className={css("kube-list-content", this.props.className)}>
                 {this._getComponentHeadingContent()}
-                {this._getComponent()}
+                {this.props.items && this.props.items.length > 0 && this._getComponent()}

```

Comment:
we are showing table only when we have data, what if we want to see headers also? may not need to fix, we have to think about it.,